pyrealm_build_data/splash/splash_run_time_series_parallel.py

```diff
@@ -0,0 +1,258 @@
+    rnn_d = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    sat = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    lv = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    pw = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    psy = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    econ = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    cond = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    eet = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    pet = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    aet = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    wn = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    ro = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    wn_spun_up = xarray.full_like(input_data.elev, fill_value=np.nan)
+
+    # Create coords tuples to pass to run_one_cell
+    coords = []
+    for lat_idx, lat in enumerate(input_data.lat):
+        for lon_idx, lon in enumerate(input_data.lon):
+            # Create the SPLASH instance
+            cell_elev = input_data["elev"][lat_idx, lon_idx]
+            coords.append(
+                (
+                    lat_idx,
+                    float(lat.data),
+                    lon_idx,
+                    float(lon.data),
+                    float(cell_elev.data),
+                )
+            )
+
+    # Use a pool of processes to run the set of cells
+    with Pool(n_cores) as pool:
+        result = pool.starmap_async(run_one_cell, zip(coords, repeat(input_data)))
+
+        results = result.get()
+
+    # Compile results
+    for res in results:
+        lat_idx, _, lon_idx, _, cell_elev = res["coords"]
+
+        if not np.isnan(cell_elev):
+            n_real += 1
+        else:
+            continue
+
+        ppfd_d[:, lat_idx, lon_idx] = res["ppfd_d"]
+        rn_d[:, lat_idx, lon_idx] = res["rn_d"]
+        rnn_d[:, lat_idx, lon_idx] = res["rnn_d"]
+        sat[:, lat_idx, lon_idx] = res["sat"]
+        lv[:, lat_idx, lon_idx] = res["lv"]
+        pw[:, lat_idx, lon_idx] = res["pw"]
+        psy[:, lat_idx, lon_idx] = res["psy"]
+        econ[:, lat_idx, lon_idx] = res["econ"]
+        cond[:, lat_idx, lon_idx] = res["cond"]
+        eet[:, lat_idx, lon_idx] = res["eet"]
+        pet[:, lat_idx, lon_idx] = res["pet"]
+        aet[:, lat_idx, lon_idx] = res["aet"]
+        wn[:, lat_idx, lon_idx] = res["wn"]
+        ro[:, lat_idx, lon_idx] = res["ro"]
+
+        wn_spun_up[lat_idx, lon_idx] = res["wn_spun_up"]
+
+    out = xarray.Dataset(
+        {
+            "ppfd_d": ppfd_d,
+            "rn_d": rn_d,
+            "rnn_d": rnn_d,
+            "sat": sat,
+            "lv": lv,
+            "pw": pw,
+            "psy": psy,
+            "econ": econ,
+            "cond": cond,
+            "eet_d": eet,
+            "pet_d": pet,
+            "aet_d": aet,
+            "wn": wn,
+            "ro": ro,
+            "wn_spun_up": wn_spun_up,
+        }
+    )
```

Comment:
I feel this could be more concise by using `df = pd.DataFrame(results)` followed by `out = df.to_xarray()`.